homeassistant/components/stream/core.py

```diff
@@ -219,11 +222,12 @@ def clear(self) -> None:
         if self._unsub is not None:
             self._unsub()
 
+    @callback
     def fire(self, _now: datetime.datetime) -> None:
         """Invoke the idle timeout callback, called when the alarm fires."""
         self.idle = True
         self._unsub = None
-        self._callback()
+        self._callback(self._hass)
```

Comment:
I think the idle function can access hass on the stream object (or as a local copy) to avoid having to plumb this through